client/hwlib/src/helpers.rs

```diff
@@ -56,4 +91,13 @@ pub(crate) mod test_utils {
         }
         bail!("ran out of places to find Cargo.lock")
     }
+
+    pub(crate) fn get_expected_json(filepath: &Path, vars: &[(&str, &str)]) -> String {
+        let mut content = read_to_string(filepath).unwrap();
+        for (k, v) in vars {
+            content = content.replace(&format!("${k}"), v);
```

Comment:
I should probably do the same in the prompting-client repo, but its probably worthwhile adding a comment here explaining what the expected template format being replaced here looks like (`$FOO` and `${FOO}`) given that the nested curly braces.